src/user/user.controller.ts

```diff
@@ -1,93 +1,74 @@
-import { Body, Controller, Delete, Get, Param, Patch, Post, Render, Request, UseGuards } from '@nestjs/common';
+import {
+    Body,
+    Controller,
+    Delete,
+    Get,
+    HttpCode,
+    Param,
+    Patch,
+    Render,
+    Request,
+    UseGuards
+} from '@nestjs/common';
 import { ApiTags } from '@nestjs/swagger';
 import { ObjectId } from 'mongoose';
 import { Roles } from '../auth/roles/roles.decorator';
 import { RolesGuard } from '../auth/roles/roles.guard';
 import { JwtAuthGuard } from '../auth/strategies/jwt/jwt-auth.guard';
+import { returnUser } from './dto/return-user.dto';
 import { UpdateUserDto } from './dto/update-user.dto';
-import { User } from './entities/user.entity';
 import { Role } from './enums/role.enum';
+import { VerifyJWTGuard } from './guards/mailVerify-jwt.guard';
 import { UserService } from './user.service';
 
 @ApiTags('user')
 @Controller('user')
 export class UserController {
-  constructor(private readonly userService: UserService) { }
+    constructor(private readonly userService: UserService) {}
 
-  @Get()
-  @UseGuards(JwtAuthGuard, RolesGuard)
-  @Roles(Role.Admin)
-  async findAll(): Promise<any> {
-    return await this.userService.findAll();
-  }
+    @Get()
+    @UseGuards(JwtAuthGuard, RolesGuard)
+    @Roles(Role.Admin)
+    async findAll(): Promise<returnUser[]> {
+        return await this.userService.findAll();
+    }
 
-  @Get('/verify')
-  @UseGuards(JwtAuthGuard, RolesGuard)
-  @Roles(Role.Admin)
-  async getVerifyByUsername(@Body() userData: { userId: ObjectId }): Promise<any> {
-    return await this.userService.findVerify(userData.userId)
-  }
+    @Get('/verify')
```

Comment:
Is the token sent with Authorization header?

Reply:
Making this a url query parameter makes more sense as this is send from a button within an email and code execution in emails is very limited!

Reply:
> Making this a url query parameter makes more sense as this is send from a button within an email and code execution in emails is very limited!

Yes, absolutely. We were wondering because there was no `/verify/:id`. But the id is getting grabbed inside the guard, isn't it? so there is no need to add as a param.